### DISCUSSION TITLE
Not storing submissions breaks dev events

### DISCUSSION THREAD

**Discussion by Larsundso** [solspace/craft-freeform#927] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
When disabling FreeForm to store submissions for a form, some dev events are no longer fired.

**Steps to reproduce**

1. Create Form
2. Disable "Store Submitted Data"
3. Register "SubmissionsService::EVENT_AFTER_SUBMIT" and/or "SubmissionsService::EVENT_BEFORE_SUBMIT" with a log to check if it runs
4. Submit Form
5. Observe Events not being fired by absence of logs

**Expected behavior**
Events should be called
or at least have this behaviour outlined in the Documentation 

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 4.5.10
- Freeform Version: 4.1.12.1
- Freeform Edition: Pro
- Fresh Install or Upgrade: Fresh

**Additional context**

**Comment by kjmartens** [comment, marked as answer]
Hi @Larsundso,

This is working as expected, as the `SubmissionsService` relies on submission data to be stored. However, you can use a different event instead.

Please try these instead: 🙂 

`Form::EVENT_SUBMIT`
https://docs.solspace.com/craft/freeform/v4/developer/events/form/#before-submitting

`Form::EVENT_AFTER_SUBMIT`
https://docs.solspace.com/craft/freeform/v4/developer/events/form/#after-submitting
